### PR TITLE
media atom es6

### PR DESCRIPTION
**Pull request by akash1810**
builds on #53

refactors media-atom-lib.js and hmac.js to be ES6 classes, making use of the database and config classes in #51 and #53

### CODE REVIEW

File: CDS/scripts/js_utils/media-atom/hmac.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        requiredConfig.forEach(c => {
+            if (! Object.keys(configObj.config).includes(c)) {
+                throw `Invalid Config. Missing ${c}`;
+            }
+        });
+
+        this.configObj = configObj;
+
+        this.sharedSecret = this.configObj.config.shared_secret;
```
Comment (clloyd) [inline CDS/scripts/js_utils/media-atom/hmac.js]: dangerous presuming `this.configObj.config` exists?

File: CDS/scripts/js_utils/media-atom/hmac.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            method: method,
+            contentType: 'application/json',
+            headers: {
+                'X-Gu-Tools-HMAC-Date': date,
+                'X-Gu-Tools-HMAC-Token': token,
+                'X-Gu-Tools-Service-Name': 'content_delivery_system'
+            }
+        };
+
+        if (Object.keys(data).length > 0) {
```
Comment (clloyd) [inline CDS/scripts/js_utils/media-atom/hmac.js]: Could we just not provide a default? and then do an `if (data)` check

File: CDS/scripts/js_utils/media-atom/hmac.js (hunk 1 above)
Comment (akash1810) [inline CDS/scripts/js_utils/media-atom/hmac.js]: Yep. The `Config` class follows a validation pattern which [catches missing properties](https://github.com/guardian/content_delivery_system/blob/aa-lint/CDS/scripts/js_utils/datastore/config.js#L62) upstream.

This does mean usage of `Config` looks a bit verbose however:

```js
const config = new Config();
config.validate()
  .then(() => {
    // do stuff
  })
  .catch(e => {
    // oh noes!
  });
```

Was tempted to do the validation in the constructor, but didn't like returning a promise in the constructor.

Happy to do this a cleaner way.

File: CDS/scripts/js_utils/media-atom/hmac.js (hunk 2 above)
Comment (akash1810) [inline CDS/scripts/js_utils/media-atom/hmac.js]: `data` is defaulted to `{}` [above](https://github.com/guardian/content_delivery_system/pull/55/files#diff-383eaa9c432acec4f4bf6efab8b0fd65R28)

`if (data)` would always be `true` now?

**Pull request merged**